### PR TITLE
Add TaskbarItem

### LINKED ISSUE #25: Add Taskbar
Add Taskbar class that allows changing taskbar icon using libunity DBus API on Linux and TaskbarList3 interface on Windows.
Setting progress bar and setting urgent state will work on both platforms, while setting `count` will be Linux-only.

### PR DESCRIPTION
**Pull request by fsobolev**
Closes #25 

Usage is very simple:

```
using var launcher = await TaskbarItem.ConnectLinuxAsync("org.kde.dolphin.desktop");
launcher?.SetUrgent(true);
launcher?.SetProgress(0.3);
```

TODO:
- [x] Windows support using ITaskbarList3 + [FlashWindowEx](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-flashwindowex) for urgent state
- [x] `SetProgressState`
- [x] `SetCount` (Linux-only)
- [x] `SetCountState` (Linux-only)
- [x] Use `Dictionary<string, object>` instead of `LauncherProperties`

`SetProgressState` will use flags like the ones in [ITaskbarList3](https://learn.microsoft.com/en-us/windows/win32/api/shobjidl_core/nf-shobjidl_core-itaskbarlist3-setprogressstate):

| Flag | Windows | Linux |
| --- | --- | --- |
| NoProgress | Stops displaying progress | Stops displaying progress |
| Indeterminate | The progress indicator does not grow in size, but cycles repeatedly along the length of the taskbar button | Same as NoProgress |
| Normal | The progress indicator grows in size from left to right | The progress indicator grows in size from left to right |
| Error | The progress indicator turns red to show that an error has occurred | Same as Normal ~~+ sets urgent state to true~~<sup>[1]</sup> |
| Paused | The progress indicator turns yellow to show that progress is currently stopped | Same as Normal |

[1] Now I think it's better to not combine progress and urgent states, let urgent state be completely separated from progress on both platforms.

### CODE REVIEW

File: Nickvision.Aura/Taskbar/TaskbarItem.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    /// Constructs TaskbarItem for Linux
+    /// </summary>
+    /// <param name="desktopFile">Desktop file name with extension</param>
+    private TaskbarItem(string desktopFile)
+    {
+        _unityLauncher = new LauncherEntry(desktopFile);
+        _disposed = false;
+    }
+    
+    private TaskbarItem(nint hwnd)
```
Comment (nlogozzo) [inline Nickvision.Aura/Taskbar/TaskbarItem.cs:29]: missing docs

File: Nickvision.Aura/Taskbar/TaskbarItem.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        _dbusConnection?.Dispose();
+        _disposed = true;
+    }
+    
+    /// <summary>
+    /// Connects to an item on the taskbar on Linux
+    /// </summary>
+    /// <param name="desktopFile">Desktop file name with extension</param>
+    /// <exception cref="PlatformNotSupportedException">Thrown if called not on Linux</exception>
+    public static async Task<TaskbarItem?> ConnectAsync(string desktopFile)
```
Comment (nlogozzo) [inline Nickvision.Aura/Taskbar/TaskbarItem.cs:74]: I'd call this `ConnectLinuxAsync`

File: Nickvision.Aura/Taskbar/TaskbarItem.cs [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Connects to an item on the taskbar on Windows
+    /// </summary>
+    /// <param name="hwnd">Window handle</param>
+    /// <exception cref="PlatformNotSupportedException">Thrown if called not on Windows</exception>
+    /// <exception cref="ArgumentException">Thrown if hwnd is IntPtr.Zero</exception>
+    public static TaskbarItem? Connect(nint hwnd)
```
Comment (nlogozzo) [inline Nickvision.Aura/Taskbar/TaskbarItem.cs:100]: and call this `ConnectWindows`

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Review by nlogozzo — APPROVED** [review]

**Comment by nlogozzo** [comment]
Will merge and release later today :)

**Comment by fsobolev** [comment]
No need to rush with the release, I can implement #24 tonight

**Comment by nlogozzo** [comment]
> I can implement https://github.com/NickvisionApps/Aura/issues/24 tonight

Ah ok, I'll wait for that then!

**Pull request merged**